plugin-flex-ts-template-v2/src/feature-library/callback-and-voicemail/custom-components/CallbackAndVoicemail/CallbackAndVoicemail.tsx

```diff
@@ -60,6 +61,19 @@ export const CallbackAndVoicemail = ({ task, allowRequeue, maxAttempts }: Callba
   const disableCallCustomerButton = disableRetryButton || workerOffline(workerActivitySid);
   const thisAttempt = callBackData?.attempts ? Number(callBackData.attempts) + 1 : 1;
 
+  const [voicemail, setVoicemail] = useState(null as FetchVoicemailResponse | null);
+
+  const fetchVoicemail = async () => {
+    setVoicemail(null);
+    if (callBackData?.recordingSid && !callBackData.isDeleted) {
+      setVoicemail(await CallbackService.fetchVoicemail(callBackData.recordingSid));
```

Comment:
it looks like if the fetch fails for whatever reason to return a valid response the screen will show "loading voicemail" indefinitely.  Perhaps we could do a check here on the response so if there is a failure to retrieve the voicemail we can inform the agent that it failed

Reply:
Added a check and an option to retry.